### PR TITLE
Proxy should allow duplicate ownKeys when returning getOwnPropertyDescriptors

### PR DESCRIPTION
**Pull request by tcare**
Fixes test262 test\built-ins\Object\getOwnPropertyDescriptors\duplicate-keys.js and OS: 8729344

Simple spec update to allow duplicate keys to be returned in Object.getOwnPropertyDescriptors by always assigning the return array value. We do not need to keep track of the count because we always remove all instances from the property map in spec code.

### CODE REVIEW

File: lib/Common/Exceptions/Throw.h [hunk 1]
```diff
@@ -208,6 +208,10 @@ namespace Js {
         {       \
             hr = GetRuntimeErrorFunc(Js::RecyclableObject::FromVar(errorObject), nullptr);   \
         }   \
+        else if (errorObject != nullptr) \
+        {  \
+            hr = JSERR_UncaughtException; \
+        }  \
```
Comment (kunalspathak) [inline lib/Common/Exceptions/Throw.h]: is this change related to proxy?

File: lib/Runtime/Library/JavascriptProxy.h [hunk 2]
```diff
@@ -185,11 +185,13 @@ namespace Js
                     {
                         targetToTrapResultMap.Add(propertyId, true);
                     }
+                }
 
-                    if (fn(propertyRecord))
-                    {
-                        trapResult->DirectSetItemAt(trapResultIndex++, element);
-                    }
+                // We explcitly allow duplicates in the results. A map is sufficient since the spec steps that remove entries
```
Comment (kunalspathak) [inline lib/Runtime/Library/JavascriptProxy.h]: >explcitly [](start = 22, length = 9)

can we have a unit test?

Comment (tcare) [inline lib/Runtime/Library/JavascriptProxy.h]: Test262 is run in nightlies and should sufficiently cover this case. I could add that one though..

File: lib/Common/Exceptions/Throw.h (hunk 1 above)
Comment (tcare) [inline lib/Common/Exceptions/Throw.h]: Nope, I need to push again.

**Comment by kunalspathak** [comment]
I think it will be good to have it so bugs are caught early in CI run itself.

Thanks,
Kunal

**Comment by akroshg** [comment]
:shipit:

**Comment by kunalspathak** [comment]
:shipit:
